task_force_bot.py

```diff
@@ -78,158 +67,496 @@ def distance_1d(val_1: Union[int, np.ndarray], val_2: Union[int, np.ndarray], si
+
+    def task_gather_halite(self, board_halite: np.ndarray, board_enemy_ships_influence_zone: np.ndarray,
+                           board_shipyards: np.ndarray, ship_position: int) -> int:
+        square_attractiveness = np.zeros((self.size, self.size))
+
+        shipyard_counts = np.column_stack(np.unique(board_shipyards[~np.isnan(board_shipyards)], return_counts=True))
+        shipyard_coordinates_array = np.column_stack(np.where(board_shipyards == self.player_id))
+
+        if shipyard_counts[shipyard_counts[:, 0] == self.player_id].size == 0:  # Player has no shipyard
+            square_attractiveness += (0.25 * board_halite
+                                      / (1 + 2 * self.distance_matrix[ship_position].reshape(self.size, -1)))
+
+        else:  # Player has at least one shipyard
+            distance_matrix_to_closest_shipyard = np.empty((self.size, self.size))
+            distance_matrix_to_closest_shipyard[:] = np.nan
+
+            for shipyard_coordinates in shipyard_coordinates_array:
+                shipyard_position = get_position(shipyard_coordinates, self.size)
+                distance_matrix_to_closest_shipyard = \
+                    np.fmin(distance_matrix_to_closest_shipyard,
+                            self.distance_matrix[shipyard_position].reshape(self.size, -1))
+
+            square_attractiveness += (0.25 * board_halite
+                                      / (1 + self.distance_matrix[ship_position].reshape(self.size, -1)
+                                         + distance_matrix_to_closest_shipyard))
+
+        # Take into account proximity of enemy ships
+        square_attractiveness *= 1 - board_enemy_ships_influence_zone
+
+        # Take into account squares that are already being targeted (evil hack)
+        square_attractiveness *= 1 - self.gather_halite_target_squares.astype(int)
+
+        gather_halite_target_position = np.argmax(square_attractiveness.flatten())
+        gather_halite_target_coordinates = get_coordinates(gather_halite_target_position, self.size)
+        self.gather_halite_target_squares[gather_halite_target_coordinates[0],
+                                          gather_halite_target_coordinates[1]] = True
+
+        return int(gather_halite_target_position)
+
+    def direction_is_free(self, ship_position: int, direction: str) -> bool:
+        ship_coordinates = get_coordinates(ship_position, self.size)
+        new_coordinates = (ship_coordinates + self.directions_dict[direction]) % self.size
+        return (~self.board_threatened_squares[new_coordinates[0], new_coordinates[1]]
+                and ~self.blocked_squares[new_coordinates[0], new_coordinates[1]])
+
+    def task_evade_enemy(self, ship_position: int) -> int:
+        free_directions = []
+        for direction in self.directions_dict:
+            if self.direction_is_free(ship_position, direction):
+                free_directions.append(direction)
+
+        try:
```

Comment:
try and except baut man eigentlich nur ein, wenn man Ereignisse gar nicht unter Kontrolle hat. Beispiel: Internetverbindung bricht ab. 

Hier kannst du doch checken, ob free_directions leer ist und wenn ja, wähle irgendeine Richtung aus.